src/lib/SX127xDriver/SX127x.cpp

```diff
@@ -518,8 +565,8 @@ void ICACHE_RAM_ATTR SX127xDriver::IsrCallback()
 {
     uint8_t irqStatus = instance->GetIrqFlags();
     instance->ClearIrqFlags();
-    if ((irqStatus & SX127X_CLEAR_IRQ_FLAG_TX_DONE))
+    if ((irqStatus & SX127X_CLEAR_IRQ_FLAG_TX_DONE) && (instance->currOpmode == SX127x_OPMODE_TX))
```

Comment:
The currOpmode checks were removed by me because they were causing deadlocks due to us expecting an interrupt and not getting it due to the opmode check. We had switched operating modes but got the interrupt for the other opmode which left our Nonce out of whack. Or code NEEDS to have the ISRs called and I think this will create a future regression when we get a Nonce slip again because the opmode didn't match when the interrupt fires.

Reply:
The behaviour you observed regarding setting one mode but getting the other should be fixed now. That was exactly the behaviour I observed before this fix, and I'm unable to make it happen anymore. I think you may have been getting hit by the same race condition before.

Reply:
The issue I was seeing before was calling the wrong handler when an interrupt fires (being in TX mode but getting an RX interrupt). Masking based on the opmode does further that fix by not generating interrupts out of order, but it adds the issue that we can now miss interrupts entirely when our code expects them to always happen.

If the TX doesn't get the TXDONE ISR because it was masked by this code, then the FHSS will not advance and everything will fall apart. That maybe can't happen because the only time it switches to opmode == RX is from the TXDONE handler, but I do worry that discarding interrupts when our code depends on them to advance counters will eventually be another mystery bug.

Reply:
These changes are like wearing a condom and a hazmat suit. I can remove the double checks if we come to a consensus.

Reply:
I just wanted to mention it as a potential source of problems because our code is pretty sensitive to skipping an ISR. I think we'll be ok on the receive side (since we frequently don't get packets), and the transmit side if we've already switched out of RX mode then it probably knows what it is doing?

We do need to add code in `RXnbISR()` though to switch the `currOpmode` back to idle if in single RX mode though. SX1280 goes back to FS and SX127x goes to STANDBY.

Reply:
Changes made, with comments from the datasheet about the modes.